src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticAnalysisContext.cs

```diff
@@ -648,12 +648,23 @@ public struct SemanticModelAnalysisContext
         /// </summary>
         public CancellationToken CancellationToken { get { return _cancellationToken; } }
 
+        /// <summary>
+        /// Optional filter span for which to compute diagnostics.
+        /// </summary>
+        internal TextSpan? FilterSpan { get; }
```

Comment:
this feels important enough that maybe it should be public?

Reply:
Likely, but the filter span concept is not applicable for batch compilation. I did mention about this in the PR description:

> Prior to this change, compiler diagnostic analyzer could only be run from the IDE for entire document span. This is primarily due to lack of an analyzer API that allows the analyzer to register a span-based semantic diagnostic callback. Though we can consider designing and adding such an analyzer API, it would be purely an IDE-only analyzer action, as analyzers are never executed for a span in batch compilation mode - this would make it difficult to justify adding such an API.